runtime/v2/runc/task/service.go

```diff
@@ -100,8 +101,8 @@ func NewTaskService(ctx context.Context, publisher shim.Publisher, sd shutdown.S
 
 // service is the shim implementation of a remote shim over GRPC
 type service struct {
-	mu          sync.Mutex
-	eventSendMu sync.Mutex
+	mu                sync.Mutex
+	processEventMUMap map[string]*sync.Mutex
```

Comment:
Would [`pkg/kmutex`](https://github.com/containerd/containerd/blob/4b7145cfd38d76bc763e3cee1210963bc6c22680/pkg/kmutex/kmutex.go) or https://github.com/moby/locker be suitable to use in place of this bespoke implementation? I've found references to both in the containerd codebase 🙈

Reply:
Ha! I had https://github.com/moby/locker open in a tab; coming from that exact thinking. ISTR some similar package was also implemented in containerd itself (but it may have had a different purpose)

Reply:
@thaJeztah @corhere ended up resorting to a different solution here, so no `kmutex` or `locker` was required 😅 still, thanks!